src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/PreviewAPIListBuilder.java

```diff
@@ -86,7 +86,12 @@ protected void handleElement(Element e) {
             }
             return new JEP(number, title, status);
         });
-        elementJeps.put(e, jep);
+        if (jep.number == 0) {
+            // Remove preview support features without a valid JEP
+            jeps.remove(feature);
```

Comment:
This feels like we are adding and removing a dummy jep to and from the map every time we encounter the language model feature. Can we just bind a dummy jep to that feature to avoid repeated processing, and only filter at the getter methods?

Reply:
I agree it's not ideal. I think the best thing is to split up the insertion into separate steps: First check if the key is contained in the map, retrieve the JEP info if it isn't, and only then decide whether to create a JEP record.

But I found another problem with this PR, so there may be some further changes coming.